src/main/java/hudson/remoting/URLish.java

```diff
@@ -16,19 +19,24 @@
 abstract class URLish {
     private URLish() {}
 
+    @CheckForNull
     abstract URL toURL() throws MalformedURLException;
 
-    static URLish from(final URL url) {
+    @Nullable
+    static URLish from(@CheckForNull final URL url) {
         if (url==null)  return null;
+        
         return new URLish() {
             @Override
+            @Nonnull
             URL toURL() {
                 return url;
             }
         };
     }
 
-    static URLish from(final File f) {
+    @Nullable
+    static URLish from(@CheckForNull final File f) {
```

Comment:
This is internal, so why not make it `@Nonnull`? Then `toURL` could also be `@Nonnull`, as could `toURLs`. If you want to signal an error, throw an exception.